numba/core/rewrites/static_getitem.py

```diff
@@ -93,6 +93,53 @@ def apply(self):
         return new_block
 
 
+@register_rewrite('after-inference')
+class RewriteStringLiteralSetitems(Rewrite):
+    """
+    Rewrite IR expressions of the kind `setitem(value=arr, index=$XX, value=)`
+    where `$XX` is a StringLiteral value as
+    `static_setitem(value=arr, index=<literal value>, value=)`.
+    """
+
+    def match(self, func_ir, block, typemap, calltypes):
+        """
+        Detect all getitem expressions and find which ones have
+        string literal indexes
+        """
+        self.setitems = setitems = {}
+        self.block = block
+        self.calltypes = calltypes
+        for inst in block.find_setitem():
+            index_ty = typemap[inst.index.name]
+            if isinstance(index_ty, types.StringLiteral):
+                setitems[inst] = (inst.index, index_ty.literal_value)
+
+        return len(setitems) > 0
+
+    def apply(self):
+        """
+        Rewrite all matching setitems as static_setitems where the index
+        is the literal value of the string.
+        """
+        new_block = self.block.copy()
+        new_block.clear()
```

Comment:
Any reason for creating a new block in this way? I saw that other passes create a block in the same way but this comes with a cost of copying all instructions. Something that you can do is:
```python
new_block = ir.block(block.scope, block.loc)
```

Reply:
I just did it because saw it somewhere else. but I think you're right.